packages/brokers/src/brokers/redis/BaseRedis.ts

```diff
@@ -187,6 +187,7 @@ export abstract class BaseRedisBroker<TEvents extends Record<string, any>>
 	public async destroy() {
 		this.streamReadClient.disconnect();
 		this.redisClient.disconnect();
+		this.removeAllListeners();
```

Comment:
This doesn't accomplish what we want here. 

For totally correct cleanup, what we have to do is keep track of everything we've `subscribe`'d to and call `unsubscribe` before destroying the actual redis clients.

The main problem with this is we don't have access to the `group`s; we should start storing a map in `subscribe` - was my first thought, but...

Looking at the implementation right now, I'm noticing that we don't really support having multiple groups set up at the same time anyway (and it doesn't make a lot sense to) - as in you can do it, but it will just break in unexpected ways because we don't track what event belongs to what group.

Let's just break the whole API instead; take `group` as a constructor option and reference it that way instead, remove it as a parameter from all the methods - that's a lot more correct and also allows us to do this here:

```ts
this.unsubscribe(...this.subscribedEvents);
```

Reply:
Sounds good! 

I think setting the group as a constructor parameter makes more sense. In my case I even created a wrapper around it that takes the group and wraps the subscribe/unsubscribe methods.

I pushed a commit LMK what you think.

Not sure what `removeAllListeners` does exactly but should we keep it as well?
There's noting else that removes the listeners

Reply:
No, cleaning up listeners I'd argue is unexpected behavior, IMO. The user should do that on their own if they want to.

Reply:
OK sound right.
Removed then